ci-operator/step-registry/upi/gcp/rhel/pre/upi-gcp-rhel-pre-ref.yaml

```diff
@@ -0,0 +1,32 @@
+ref:
+  as: upi-gcp-rhel-pre
+  from: libvirt-installer
+  grace_period: 10m
+  commands: upi-gcp-rhel-pre-commands.sh
+  resources:
+    requests:
+      cpu: 10m
+      memory: 100Mi
+  env:
+    - name: HOME
+      default: /tmp/secret
+    - name: NSS_WRAPPER_PASSWD
+      default: /tmp/secret/passwd
+    - name: NSS_WRAPPER_GROUP
+      default: /tmp/secret/group
+    - name: NSS_USERNAME
+      default: rhel8user
+    - name: NSS_GROUPNAME
+      default: rhel8user
+    - name: GOOGLE_COMPUTE_IMAGE_FAMILY
+      default: rhel-8
+    - name: GOOGLE_COMPUTE_IMAGE_NAME
+      default: ''
+    - name: GOOGLE_COMPUTE_IMAGE_PROJECT
+      default: rhel-cloud
+    - name: GOOGLE_COMPUTE_MACHINE_TYPE
+      default: e2-standard-4
```

Comment:
Is this machine slower than the current one?

Reply:
It appears so - 8 vs 4 vCPUs, each vCPU is 4GB of memory:
```
| Machine types | vCPUs* | Memory (GB) | Max number of persistent disks (PDs)† | Max total PD size (TB) | Local SSD | Maximum egress bandwidth (Gbps)‡ |
| e2-standard-2 |      2 |           8 |                                   128 |                    257 | No        |                                4 |
| e2-standard-4 |      4 |          16 |                                   128 |                    257 | No        |                                8 |
| e2-standard-8 |      8 |          32 |                                   128 |                    257 | No        |                               16 |
```

I'd say we should be as close to "advertised" requirements as possible so we catch performance problems and also HW requirements creep.
On the other hands, `openshift-tests` might not be close to reference workload.

Maybe we could have a performance/resource-usage focused periodic job?
Use `e2-standard-{2,4}`, run `openshift-tests` against it and measure and make a report out of bunch of stuff:
- how much disk space we use for images and other persisted artifacts in /var/
- prometheus + node exporter

Make it usable with Vadim's fake KAS and loki apps?

Reply:
>  On the other hands, openshift-tests might not be close to reference workload.

Correct, the tests spin up dozens of workloads simultaneously, so reducing the hw will at best just extend the runtime of the tests, at worst cause hard to debug errors.

Reply:
What I meant in my comment was if this makes the tests run slower or if there is no difference at all. If its the latter we are wasting money.

It is true that we should run as close to the advertised requirements as possible, but `openshift-tests` are not a regular use case.
Not only is it harsh on the control plane and scheduler, it is also trying to test conformance against a set of requirements as quick as possible.
I would prioritize runtime efficiency because these tests run on every PR, so the faster the better. Same goes for stuff we add to monitor, like prometheus.

Reply:
Yeah, I see what you mean.  I'll revert to the higher powered machine and we can do testing later at our convenience.